ape_hardhat/providers.py

```diff
@@ -382,9 +382,25 @@ def get_transaction_trace(self, txn_hash: str) -> Iterator[TraceFrame]:
         Returns:
             Iterator(TraceFrame): Transaction execution trace object.
         """
-        logs = self._make_request("debug_traceTransaction", [txn_hash]).structLogs
-        for log in logs:
-            yield TraceFrame(**log)
+
+        for frame in self._make_request("debug_traceTransaction", [txn_hash]).structLogs:
+            yield TraceFrame(**frame)
+
+    def get_calltree_from_trace(
+        self, txn_hash: str, show_internals=False, **root_node_kwargs
+    ) -> CallTreeNode:
+        txn = self.get_transaction(txn_hash=txn_hash)
+        root_node_kwargs["gas_cost"] = txn.gas_used
+        root_node_kwargs["gas_limit"] = txn.gas_limit
+        root_node_kwargs["address"] = txn.receiver
+        # TODO: logic for calldata from a transaction
+        root_node_kwargs["calldata"] = getattr(txn, "data", b"")
+        # TODO: logic for CallType
+        root_node_kwargs["call_type"] = CallType.MUTABLE
```

Comment:
One way that Brownie computes coverage is by turning `eth_call` calls (e.g. view calls) into `eth_sendTransaction` calls by snapshotting and reverting. It will then use the `debug_traceTransaction` mechanism to get the coverage data out. So... We can use this here.